### ISSUE TITLE
Suggestion: semantic html elements

### ISSUE THREAD

**Issue by samuelstroschein** [saadeghi/daisyui#217]
DaisyUI would greatly improve the readability of code if the "component" classes would be HTML tags that can be styled with the utility classes.

**Short example**:
```HTML
  <div class="collapse-title text-xl font-medium">
    I open with focus
  </div> 

  <!-- collapse-title is a class but should be the HTML tag -->

  <collapse-title class="text-xl font-medium">
    I open with focus
  </collapse-title> 
```

**Long example Instead of this**:
```HTML

<div tabindex="0" class="collapse w-96"> 
  <div class="collapse-title text-xl font-medium">
    I open with focus
  </div> 
  <div class="collapse-content"> 
    <p>Collapse content reveals with focus. If you add a checkbox, you can control it using checkbox instead of focus. Or you can force-open/force-close using 
      <span class="badge badge-outline">collapse-open</span> and 
      <span class="badge badge-outline">collapse-close</span> classes.
    </p>
  </div>
</div> 

``` 

**This**
```HTML
<collapse tabindex="0" class="w-96">
	<collapse-title class="text-xl font-medium">I open with focus</collapse-title>
	<collapse-content class="">
		<p>
			Collapse content reveals with focus. If you add a checkbox, you can control it using checkbox
			instead of focus. Or you can force-open/force-close using
			<badge class="badge-outline">collapse-open</badge> and
			<badge class="badge-outline">collapse-close</badge> classes.
		</p>
	</collapse-content>
</collapse>


```

> TL;DR – we finally have a way to create our own HTML elements and use them in any framework we want!
> 
> Thanks to the popularity of frameworks such as Angular, Vue, and React, component-driven development has become a part of our every day lives. Components help us encapsulate styles and behaviors into reusable building blocks. They make a lot of sense in terms of design, development, and testing.
> 
> Unfortunately, framework-specific components fail us in a number of ways:
> 
> You can only use them in the framework they're designed for 🔒
> Their lifespan is limited to that of the framework's ⏳
> New frameworks/versions can lead to breaking changes, requiring substantial effort to update components 😭
> Web components solve these problems. They're supported by all modern browsers, they're framework-agnostic, and they're part of the standard, so we know they'll be supported for many years to come.
TL;DR on web components  from [shoelace](https://shoelace.style/)

**Comment by saadeghi** [comment]
@samuelstroschein  
Thanks for the detailed comment.  
There's one big problem about this approach:  
When there's a `.btn` class, we can use that class everywhere: `<a class="btn">`, `<button class="btn">`, `<label class="btn">`,... but when we use `<btn>` it's useless.   

Tailwind CSS provides **utility classes** and daisyUI adds **component classes** on top. So we're still using a **utility first** system.

**Comment by samuelstroschein** [comment]
Adding default styling to semantic HTML tags does not entail removing the component classes. I meant that DaisyUI should apply (maybe opt-in) the component classes by default to those HTML elements. 

```HTML
<btn class="btn-outline">Semantic</btn>

<div class="btn btn-outline">Works as well</div>
```

**Comment by saadeghi** [comment]
This doubles the size of CSS files (in both source and package) but I'm not sure what problem is being fixed here 🤷‍♂️

**Comment by samuelstroschein** [comment]
Why would it require duplicative CSS files? TailwindCSS allows [adding base styles](https://tailwindcss.com/docs/adding-base-styles) to HTML elements. The only additional step that is required is defining the custom HTML element tags such as `<collapse-title></collapse-title>` etc but they are more less just `divs` with a semantic name.

> I'm not sure what problem is being fixed here 🤷‍♂️

It's a code readability enhancement. Instead of scaffolding meaningless elements like `<div>` all over the markup, that only get meaning with the component class e.g. `<div class="collapse-title">`, the relatively new introduction of custom HTML tags  would restore semantics e.g. `<collapse-title>`. 

### Why is this beneficial?
Look at the code from the [cards example](https://daisyui.com/demos/cards). Congrats, everything is a meaningless `div`. A developer now needs to read the class first to understand which component belongs to which `div`. 

<img width="1398" alt="Screenshot 2021-09-27 at 14 59 48" src="https://user-images.githubusercontent.com/35429197/134913127-1f14c5a5-9139-49fb-9181-d12872bdc891.png">

 That is not too problematic when using inspect but by just looking at the source code, don't you agree that having a semantic tag name (which is a div under the hood having the component class applied by default) would improve the readability?

**Comment by TotalLag** [comment]
I came here to request something similar as I am exploring the examples and wondered why the navigation elements are all <div>'s and why <nav> wasn't being used.
